haystack/nodes/prompt/prompt_node.py

```diff
@@ -326,6 +330,10 @@ def __init__(
             use_auth_token=self.use_auth_token,
             model_kwargs=model_input_kwargs,
         )
+        # This is how the default max_length is determined for Text2TextGenerationPipeline shown here
+        # https://huggingface.co/transformers/v4.6.0/_modules/transformers/pipelines/text2text_generation.html
+        # max_length must be set otherwise HFLocalInvocationLayer._ensure_token_limit will fail.
+        self.max_length = max_length or self.pipe.model.config.max_length
```

Comment:
Here is where I set `max_length` in `HFLocalInvocationLayer`. It requires the model to already be loaded to be able to set a default value if the user passes `None` as the `max_length`.